File: example/src/main/java/com/github/vivchar/example/BaseScreenFragment.java
package com.github.vivchar.example;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

/**
 * Created by Vivchar Vitaly on 12/28/17.
 */

public class BaseScreenFragment extends Fragment {


	@NonNull
	public UIRouter getUIRouter() {
		return ((MainActivity)getActivity()).getUIRouter();
	}

	@NonNull
	public OptionsMenuController getMenuController() {
		return ((MainActivity)getActivity()).getMenuController();
	}
}

File: example/src/main/java/com/github/vivchar/example/BaseScreenFragment.kt
package com.github.vivchar.example

import androidx.fragment.app.Fragment

/**
 * Created by Vivchar Vitaly on 12/28/17.
 */
open class BaseScreenFragment : Fragment() {
	val router: UIRouter get() = (activity as MainActivity?)!!.router!!
	val menuController: OptionsMenuController get() = (activity as MainActivity?)!!.menuController!!
}